R/module_data_summary.R

```diff
@@ -163,83 +177,86 @@ get_filter_overview <- function(teal_data) {
+get_filter_overview <- function(current_data, initial_data, dataname, subject_keys) {
+  if (inherits(current_data, c("data.frame", "DataFrame", "array", "Matrix", "SummarizedExperiment"))) {
+    get_filter_overview_array(current_data, initial_data, dataname, subject_keys)
+  } else if (inherits(current_data, "MultiAssayExperiment")) {
+    get_filter_overview_MultiAssayExperiment(current_data, initial_data, dataname)
   } else {
-    data.frame(
-      dataname = dataname,
-      obs = NA,
-      obs_filtered = NA,
-      subjects = NA,
-      subjects_filtered = NA
-    )
+    data.frame(dataname = dataname)
   }
 }
 
 #' @rdname module_data_summary
-get_object_filter_overview_array <- function(filtered_data, # nolint: object_length.
-                                             unfiltered_data,
-                                             dataname,
-                                             subject_keys) {
+get_filter_overview_array <- function(current_data, # nolint: object_length.
+                                      initial_data,
+                                      dataname,
+                                      subject_keys) {
   if (length(subject_keys) == 0) {
     data.frame(
       dataname = dataname,
-      obs = ifelse(!is.null(nrow(unfiltered_data)), nrow(unfiltered_data), NA),
-      obs_filtered = nrow(filtered_data),
-      subjects = NA,
-      subjects_filtered = NA
+      obs = if (!is.null(initial_data)) {
+        sprintf("%s/%s", nrow(current_data), nrow(initial_data))
+      } else {
+        nrow(current_data)
+      }
     )
   } else {
     data.frame(
       dataname = dataname,
-      obs = ifelse(!is.null(nrow(unfiltered_data)), nrow(unfiltered_data), NA),
-      obs_filtered = nrow(filtered_data),
-      subjects = nrow(unique(unfiltered_data[subject_keys])),
-      subjects_filtered = nrow(unique(filtered_data[subject_keys]))
+      obs = if (!is.null(initial_data)) {
+        sprintf("%s/%s", nrow(current_data), nrow(initial_data))
+      } else {
+        nrow(current_data)
+      },
+      subjects = if (!is.null(initial_data)) {
+        sprintf("%s/%s", nrow(unique(current_data[subject_keys])), nrow(unique(initial_data[subject_keys])))
+      } else {
+        nrow(unique(current_data[subject_keys]))
+      }
     )
   }
 }
 
 #' @rdname module_data_summary
-get_object_filter_overview_MultiAssayExperiment <- function(filtered_data, # nolint: object_length, object_name.
-                                                            unfiltered_data,
-                                                            dataname) {
-  experiment_names <- names(unfiltered_data)
+get_filter_overview_MultiAssayExperiment <- function(current_data, # nolint: object_length, object_name.
+                                                     initial_data,
+                                                     dataname) {
+  experiment_names <- names(initial_data)
```

Comment:
Having `initial_data` here creates problem with MAE that's added during transforms.

```suggestion
  experiment_names <- names(current_data)
```

Reply:
Here's a test spec for this case

```r
testthat::test_that("summary table displays MAE dataset added in transforms", {
  data <- within(teal.data::teal_data(), {
    iris <- iris
    mtcars <- mtcars
    foo <- identity
  })
  shiny::testServer(
    app = srv_teal,
    args = list(
      id = "test",
      data = data,
      modules = modules(module("module_1", server = function(id, data) data, datanames = "all", transformers = list(
        teal_transform_module(
          server = function(id, data) {
            reactive({
              data() |> within({
                withr::with_package("MultiAssayExperiment", {
                  data("miniACC", package = "MultiAssayExperiment", envir = environment())
                })
              })
            })
          }
        )
      )))
    ),
    expr = {
      session$setInputs("teal_modules-active_tab" = "module_1")
      session$flushReact()
      
      testthat::expect_equal(
        module_summary_table(output, "module_1"),
        data.frame(
          "Data Name" = c(
            "foo", "iris", "miniACC", "- RNASeq2GeneNorm", "- gistict", 
            "- RPPAArray", "- Mutations", "- miRNASeqGene", "mtcars"
          ),
          Obs = c("", "150/150", "", "198", "198", "33", "97", "471", "32/32"),
          Subjects = c(NA_integer_, NA_integer_, 92, 79, 90, 46, 90, 80, NA_integer_),
          check.names = FALSE
        )
      )
    }
  )
})
```